javascript-javascript1-week3.js

```diff
@@ -0,0 +1,181 @@
+// Its warmup time!
+// Item array removal
+
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+];
+const nameToRemove = names.splice(1, 1);
+console.log(nameToRemove);
+
+console.log(names); // ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+// // When will we be there?
+
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+};
+
+const travelTime = calculateTime(travelInformation);
+console.log(travelTime); // 8 hours and 38 minutes
+
+function calculateTime(){
+    const time = travelInformation.destinationDistance / travelInformation.speed;
+    const hours = Math.floor(time);
+    const minutes= Math.floor((time - hours) * 60);
+    return `${hours} hours and ${minutes} minutes`;
+}
+
+// // Series duration of my life
+
+const seriesDurations = [
+    {
+        title: "MasterChef",
+        days: 22,
+        hours: 6,
+        minutes: 56,
+    },
+    {
+        title: "Locke & key",
+        days: 2,
+        hours: 12,
+        minutes: 23,
+    },
+    {
+        title: "The Vampire Diaries",
+        days: 7,
+        hours: 14,
+        minutes: 56,
+    },
+];
+
+function durationInMinutes(days, hours, minutes) {
+    return days * 24 * 60 + hours * 60 + minutes;
+}
+function logOutSeriesText() {
+    const averageLifeSpanYears = 80;
+    const totalLifeMinutes = averageLifeSpanYears * 365.25 * 24 * 60;
+
+    let totalMinutesWatched = 0;
+
+    seriesDurations.forEach(series => {
+        const seriesMinutes = durationInMinutes(series.days, series.hours, series.minutes);
+        totalMinutesWatched += seriesMinutes;
+
+        const lifePercentage = (seriesMinutes / totalLifeMinutes) * 100;
+        console.log(`${series.title} took ${lifePercentage.toFixed(2)}% of my life`);
+    });
+
+    const totalLifePercentage = (totalMinutesWatched / totalLifeMinutes) * 100;
+    console.log(`In total that is ${totalLifePercentage.toFixed(2)}% of my life`);
+}
+
+logOutSeriesText();
+
+// //// Smart-ease - Back to the basics!
+// //NOnoN0nOYes
+
+const notes = [];
+
+function saveNote(content, id){
+    notes.push({content, id});
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+// Get a note
+
+function getNote(id) {
+
+    if (typeof id !== 'number' || id === undefined){
+        console.error('Invalid ID: ID must be a number.');
+        return null;
+    }
+
+    for (let note of notes) {
+        if (note.id === id) {
+            return note;
+        }
+    }
+
+        console.error(`Note with ID ${id} not found.`);
+        return null;
+}
```

Comment:
I'd recommend you to install the Prettier VS Code extension and set it up to format your code on save. It can make the code a bit more readable and can save you some time from trying to align things.

Here is a good article explaining how to set it up - https://www.digitalocean.com/community/tutorials/how-to-format-code-with-prettier-in-visual-studio-code